toolkit/tools/imagegen/configuration/networkconfig.go

```diff
@@ -0,0 +1,368 @@
+
+	if netMask != "" && ip != "" {
+		stringMask := net.IPMask(net.ParseIP(networkData.NetMask).To4())
+		cidrPrefix, _ := stringMask.Size()
+		ipAddr := networkData.Ip + "/" + strconv.Itoa(cidrPrefix)
+
+		networkSection = networkSection + ipField + ipAddr + "\n"
+	}
+
+	// Update Gateway
+	gateway := strings.Trim(networkData.GateWay, " ")
+	if gateway != "" {
+		networkSection = networkSection + gateWayField + networkData.GateWay + "\n"
+	}
+
+	// Update Nameserver
+	for _, nameserver := range networkData.NameServer {
+		dnsSetting := dnsField + nameserver + "\n"
+		networkSection = networkSection + dnsSetting
+	}
+
+	err = file.Append(networkSection, fileName)
+	if err != nil {
+		logger.Log.Infof("Failed to write [Network] section: %s", err)
+	}
+
+	return
+}
+
+func createNetworkConfigFile(installChroot *safechroot.Chroot, networkData Network, deviceName string) (err error) {
+	const (
+		networkFileDir = "/etc/systemd/network/"
+		filePrefix     = "10-"
+	)
+
+	if exists, ferr := file.DirExists(networkFileDir); ferr != nil {
+		logger.Log.Errorf("Error accessing /etc/systemd/network/")
+		err = ferr
+		return
+	} else if !exists {
+		err = fmt.Errorf("/etc/systemd/network/: no such path or directory")
+		return
+	}
+
+	networkFileName := networkFileDir + filePrefix + networkData.BootProto + "-" + deviceName + ".network"
+	if exists, _ := file.PathExists(networkFileName); exists {
+		return fmt.Errorf("Network file (%s) already exists", networkFileName)
+	}
+
+	err = file.Create(networkFileName, 0644)
+	if err != nil {
+		logger.Log.Errorf("Error creating file %s: %s", networkFileName, err)
+		return
+	}
+
+	// Write the [match] field
+	err = populateMatchSection(networkData, networkFileName, deviceName)
+	if err != nil {
+		return
+	}
+
+	// Write the [Network] field
+	err = populateNetworkSection(networkData, networkFileName)
+	if err != nil {
+		return
+	}
+
+	// Determine whether to activate this device on boot
+	if networkData.OnBoot {
+		installNetworkFile := filepath.Join(installChroot.RootDir(), networkFileName)
+		err = file.Copy(networkFileName, installNetworkFile)
+	}
+
+	return
+}
+
+func updateHostName(hostName string) (err error) {
+	const squashErrors = false
+
+	hostname := strings.TrimSpace(hostName)
+	if hostname != "" {
+		err = shell.ExecuteLive(squashErrors, "hostnamectl", "set-hostname", hostname)
+		if err != nil {
+			logger.Log.Errorf("Unable to set hostname: %s", err)
+		}
+	}
+
+	return
+}
+
+// ConfigureNetwork performs network configuration during the kickstart unattended installation process
+func ConfigureNetwork(installChroot *safechroot.Chroot, systemConfig SystemConfig) (err error) {
+	const squashErrors = false
+	var dnsUpdate bool
+
+	for _, networkData := range systemConfig.Networks {
+		deviceName, err := checkNetworkDeviceAvailability(networkData)
+		if err != nil || deviceName == "" {
+			logger.Log.Errorf("Could not find matching network device in the system")
```

Comment:
Matching to what? Since we're going through multiple network settings, please add more details to the log, so the user can identify the faulty config.